### ISSUE TITLE
Feature request: Allow parameters for periodic jobs

### ISSUE THREAD

**Issue by jalkanen** [gocraft/work#12]
Currently periodic jobs cannot have parameters. It would be very useful to be able to do something like enqueuer.PeriodicallyEnqueue(name, params)

Also, it's a bit inconsistent that periodical enqueues are a part of the WorkerPool, but non-periodical functions are in Enqueuer.

**Comment by cypriss** [comment]
@jalkanen Do you have a specific use-case mind here? I ask because I have never had a use for specific parameters, because the parameters need to be static. So you'd just be passing it the same thing anyway.

The reason it's part of the WorkerPool and not the enqueuer is because WorkerPools already have start/stop semantics, and in my mind are "clustered". This lets them coordinate with the other worker pools so that across 10 workers, we're only enqueueing the periodic job once. On the other hand, Enqueuers are simple clients that are not clustered and don't run in the background.

**Comment by jalkanen** [comment]
Yes.  I'm running the same script with multiple different parameters on varying, user-selectable intervals.  I can of course manipulate the queue name for the params, but that seems a bit... burdensome.

For example: I have a "run_test" which will be run against "server1", "server2", "server3" on regular intervals. Unfortunately as the schedules for each are different, I can't just have a single "run_test" job for all of them.

Static params are fine in this case; it would just make coding easier when I wouldn't have to deconstruct the params from the job name...

**Comment by cypriss** [comment]
Got it. That use-case makes sense. I'd be open to a PR that implements params.

On the other hand, depending on how many servers you have, it could be reasonable to just have jobs like "run_tests_server1", etc. That's ok for like 5 servers. For 100 it's very burdensome.

**Comment by matt-harvey** [comment]
Hey, I've raised a PR to implement this feature: https://github.com/gocraft/work/pull/119.
